great_expectations/experimental/datasources/data_asset/data_connector/s3_data_connector.py

```diff
@@ -13,19 +13,26 @@
     FilePathDataConnector,
 )
 
-try:
-    import boto3
-except ImportError:
-    boto3 = None
-
-
 if TYPE_CHECKING:
     from great_expectations.core.batch import BatchDefinition
 
 
 logger = logging.getLogger(__name__)
 
 
+try:
+    import boto3
+    import botocore
+    from botocore.client import BaseClient
+except ImportError:
+    logger.debug(
+        "Unable to load boto3 or botocore; install optional boto3 and botocore dependencies for support."
+    )
+    boto3 = None
+    botocore = None
+    BaseClient = None
+
+
```

Comment:
I really dislike this style of importing and then setting things to null if they aren't importable.
Are any of these actually used at runtime?
```python
if TYPE_CHECKING:
    from great_expectations.core.batch import BatchDefinition
    import boto3
    import botocore
    from botocore.client import BaseClient
```

Reply:
@Kilo59 I agree with you (please see the latest version -- we still need an isolated import for `boto3`, but I was able to put the other ones into `TYPE_CHECKING`, where they belong).  Thank you.